### ISSUE TITLE
charging_stations.json causes rapidly repeating forced update

### ISSUE THREAD

**Issue by denisfrench** [SAIC-iSmart-API/saic-python-mqtt-gateway#113]
Thank you to all involved in the development of this. I have the HA add-on successfully up and running with my MG4.

When I integrate the gateway directly with my [OpenEVSE ](https://www.openevse.com/) charge controller using charging_stations.json, I have found the gateway continiously force-updates the vehicle status every 30 seconds or so when it detects the EVSE is charging via MQTT.

Log extract is as follows:
```
2023-12-07 14:36:26,923 [  INFO  ] Refreshing vehicle status succeeded... - __main__
2023-12-07 14:36:27,008 [  INFO  ] Setting refresh mode to periodic - vehicle
2023-12-07 14:36:27,009 [  INFO  ] Updating vehicle status - __main__
2023-12-07 14:36:28,805 [  INFO  ] Setting refresh mode to force - vehicle
2023-12-07 14:36:43,634 [  INFO  ] Updating charging status - __main__
2023-12-07 14:37:00,221 [  INFO  ] Setting charging query interval in vehicle handler for VIN LSJ<redacted> to 0:39:56 - vehicle
2023-12-07 14:37:00,222 [  INFO  ] Refreshing vehicle status succeeded... - __main__
2023-12-07 14:37:00,307 [  INFO  ] Setting refresh mode to periodic - vehicle
2023-12-07 14:37:00,308 [  INFO  ] Updating vehicle status - __main__
2023-12-07 14:37:17,001 [  INFO  ] Updating charging status - __main__
2023-12-07 14:37:29,277 [  INFO  ] Setting refresh mode to force - vehicle
2023-12-07 14:37:33,621 [  INFO  ] Setting charging query interval in vehicle handler for VIN LSJ<redacted> to 0:39:56 - vehicle
2023-12-07 14:37:33,622 [  INFO  ] Refreshing vehicle status succeeded... - __main__
2023-12-07 14:37:33,689 [  INFO  ] Setting refresh mode to periodic - vehicle
2023-12-07 14:37:33,691 [  INFO  ] Updating vehicle status - __main__
2023-12-07 14:37:50,320 [  INFO  ] Updating charging status - __main__
2023-12-07 14:38:07,421 [  INFO  ] Setting charging query interval in vehicle handler for VIN LSJ<redacted> to 0:39:56 - vehicle
```

I can see the charging query interval is being successfully set and the mode set to periodic, but is then immediately set back to forced. The forced update period coincides with the period that my OpenEVSE updates the charging status via MQTT.

My Python is pretty rudamentary but the code here; https://github.com/SAIC-iSmart-API/saic-python-mqtt-gateway/blob/8eba0a30a2e2d0360397c75009ebb367ec1bfcf3/mqtt_publisher.py#L97C5-L103 appears to set the mode to forced every time a message indicating the the EVSE is charging is received.

I suspect the charging_stations.json implementation may have been tested with an EVSE that only updates the MQTT charge status when it changes, as opposed to my OpenEVSE that updates the status periodically regardless of whether it has changed or not.

Probalby the code section needs to be modified to set forced mode only once per session, or only when the charging state first changes to 'charging', or only if the charging query interval hasn't just been set. 

Unfortunately this issue renders the use of charging detection via MQTT unusable for me, as activating the vehicle turns my wireless Android Auto adaptor on, connecting my phone and bumping any other wifi or bluetooth connection. Besides it can't be good for whatever hums and clicks (relays?) every time the vehicle activates, or the 12v battery.

Fixing this myself is well beyond my Python skill level and understanding of how the code operates; grateful for anyone else that can take a look.

**Comment by denisfrench** [comment]
I've downloaded a local copy of the HA addon, but am struggling to get it to point to the 0.5.0-rc branch of the Python gateway repository.

I'm new to HAOS/Docker, having recently migrated from running HA-core for ages. Am I going about this the wrong way?!?

**Comment by denisfrench** [comment]
Gave up on updating to 0.5.0-rc within the add-on, but got the Python gateway running without a container on an RPi.

Just waiting for some sunshine tomorrow to see how it goes...

**Comment by Tonno87** [comment]
Both HA Add-ons need an image and a tag like 0.4.5 to grab the image from docker hub. But the 5.0.0rc isn't available on docker hub, so there is no easy way to modify the add-on.

**Comment by tosate** [comment]
Take a look at version 0.4.6 please!

**Comment by Tonno87** [comment]
@denisfrench Update to 0.4.6 is out: https://github.com/SAIC-iSmart-API/saic-home-assistant-addon/pull/45

Go to home assistant settings and check for updates.

@tosate I will test the fix with evcc too.

**Comment by denisfrench** [comment]
Tested and issue fixed for both 0.5.0-rc running standalone, and 0.4.6 running within the HA addon.

I'll wait for @Tonno87 to report back before closing, but thanks heaps for the rapid fix 🙂

**Comment by denisfrench** [comment]
From my testing the original issue is fixed, but I think there may be something funny going on with the charging state not being updated after the _second_ charge session in a row starts.

Will close this, and open a new issue once I have tested properly and collected some logs.

**Comment by denisfrench** [comment]
Hi @tosate;

I went to buy you a coffee/beer to thank you for fixing this, but couldn't see a link on your profile.

Do you have a Patreon or PayPal account or some such?
